msm/src/mvlookup.rs

```diff
@@ -260,20 +436,31 @@ pub mod prover {
                     // Iterate over the rows
                     for j in 0..domain.d1.size {
                         // Iterate over individual columns (i.e. f_i and t)
-                        for f_i in f.iter() {
+                        for (i, f_i) in f.iter().enumerate() {
                             let MVLookup {
                                 numerator: _,
                                 table_id,
                                 value,
                             } = &f_i[j as usize];
-                            // x + r * y + r^2 * z + ... + r^n table_id
+                            // Compute r * x_{1} + r^2 x_{2} + ... r^{N} x_{N}
                             let combined_value: G::ScalarField =
-                                value.iter().rev().fold(G::ScalarField::zero(), |x, y| {
-                                    x * vector_lookup_combiner + y
-                                }) * vector_lookup_combiner
-                                    + table_id.to_field::<G::ScalarField>();
+                                value.iter().rev().fold(G::ScalarField::zero(), |acc, y| {
+                                    acc * vector_lookup_combiner + y
+                                }) * vector_lookup_combiner;
+                            // add table id
+                            let combined_value =
+                                combined_value + table_id.to_field::<G::ScalarField>();
+
+                            // If last element and fixed lookup tables, we keep
```

Comment:
I don't think I get this step

Reply:
It is a way to save the fixed lookup tables